instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/engine/BeanAttributeExtractor.java

```diff
@@ -42,31 +44,59 @@ public class BeanAttributeExtractor implements MetricAttributeExtractor {
    * @throws IllegalArgumentException if the attribute name is malformed
    */
   public static BeanAttributeExtractor fromName(String rawName) {
-    if (rawName.isEmpty()) {
-      throw new IllegalArgumentException("Empty attribute name");
+    List<String> segments = splitByDot(rawName);
+    String baseName = segments.remove(0);
+    if (segments.isEmpty()) {
+      return new BeanAttributeExtractor(baseName);
     }
+    return new BeanAttributeExtractor(baseName, segments.toArray(new String[segments.size()]));
+  }
 
-    // Check if a CompositeType value is expected
-    int k = rawName.indexOf('.');
-    if (k < 0) {
-      return new BeanAttributeExtractor(rawName);
-    }
+  /*
+   * Split a given name into segments, assuming that a dot is used to separate the segments.
+   * However, a dot preceded by a backslash is not a separator.
+   */
+  private static List<String> splitByDot(String rawName) {
+    List<String> components = new ArrayList<>();
+    try {
+      StringBuilder currentSegment = new StringBuilder();
+      boolean escaped = false;
+      for (int i = 0; i < rawName.length(); ++i) {
+        char ch = rawName.charAt(i);
+        if (escaped) {
+          // No matter what it is, it becomes a part of the attribute name
```

Comment:
I sort of prefer only allowing `.` and `\` to be escaped

Reply:
Thanks for the review! Yes, escaping makes sense only for these two characters. But what should we do if someone escapes another character? Rejecting the JMX rule input seems like a very harsh reaction. Dropping the escaped character (not `.` or `\`) seems even worse.

Reply:
> Rejecting the JMX rule input

this would be my preference, and my experience with other escaping rules like JSON's

Reply:
ok, I'll make this change.